documentation/source/Usecases/mongodb.rst

```diff
@@ -0,0 +1,431 @@
+
+MongoDB
+=========
+Running MongoDB on OpenEBS
+----------------------------
+This section provides detailed instructions which allow you to perform the following tasks.
+
+* Run a MongoDB statefulset on OpenEBS storage in a Kubernetes cluster
+* Generate standard OLTP load on MongoDB using a custom Sysbench tool
+* Test data replication across the MongoDB instances
+
+Prerequisites
+---------------
+Prerequisites include the following:
+
+* A fully configured Kubernetes cluster (versions 1.6.3/4/6,1.7.0 & 1.7.5 have been tested) with Kubernetes master and at least one Kubernetes node. This can be created on cloud platforms like Google Cloud Engine, on-premise virtual machines (vagrant/VMware/Hyper-V) or Baremetal boxes.
+
+**Note:**
+
+OpenEBS recommends using a 3-node cluster, with one master and two nodes, as this aids creation of storage replicas on separate nodes. This is helpful to maintain redundancy and data availability.
+::
+    
+    test@Master:~$ kubectl get nodes
+      NAME     STATUS    AGE       VERSION
+      host01   Ready     18h       v1.6.3
+      host02   Ready     18h       v1.6.3
+      master   Ready     18h       v1.6.3
+
+* Sufficient resources on the nodes to host the OpenEBS storage pods and application pods. This includes sufficient disk space. In this example, physical storage for the volume containers will be created from the local storage.* 
+
+* ISCSI support on the nodes is required to consume the iSCSI target exposed by the OpenEBS volume container. In ubuntu, the iSCSI initiator can be installed using the following procedure.
+::
+
+    sudo apt-get update
+    sudo apt-get install open-iscsi
+    sudo service open-iscsi restart
+
+**Verify that iSCSI is configured using the following commands.**
+::
+
+    sudo cat /etc/iscsi/initiatorname.iscsi
+    sudo service open-iscsi status
+
+* Install the following dependent packages to run mongodb-integrated sysbench input/output tool on any one of the Kubernetes nodes.
+  ::
+
+    sudo apt-get install <packagename>:
+
+    make
+    libsas12-dev
+    libssl-dev
+    libmongoc-dev
+    libbson-dev
+
```

Comment:
We need to include the following packages also to the above list : 

- automake
- libmysqlclient-dev
- libtool

Reply:
Done!